py/torch_tensorrt/dynamo/conversion/impl/pad.py

```diff
@@ -0,0 +1,285 @@
+import copy
+from typing import Optional, Sequence, Union
+
+import torch
+import torch_tensorrt.dynamo.conversion.impl as impl
+from torch.fx.node import Target
+from torch_tensorrt.dynamo._SourceIR import SourceIR
+from torch_tensorrt.dynamo.conversion._ConversionContext import ConversionContext
+from torch_tensorrt.fx.converters.converter_utils import has_dynamic_shape
+from torch_tensorrt.fx.types import TRTTensor
+
+
+def constant_padNd(
+    ctx: ConversionContext,
+    target: Union[Target, str],
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    pad: Sequence[int],
+    value: Union[int, float] = 0,
+) -> TRTTensor:
+    """
+    Note: IPaddingLayer is deprecated in TensorRT 8.2 and will be removed in TensorRT 10.0.
+    Use ISliceLayer to pad the tensor, which supports new non-constant, reflects padding
+    mode and clamp, and supports padding output with dynamic shape.
+    """
+    if has_dynamic_shape(input.shape):
+        assert input.shape[1] != -1, "Channel dim can't be dynamic for padding."
+
+    # Implement constant padding via concat
+    curr_dim = len(input.shape) - 1
+
+    for i in range(0, len(pad), 2):
+        input_shape = list(input.shape)
+
+        pre_pad = pad[i]
+        post_pad = pad[i + 1]
+        pre_pad_shape = copy.deepcopy(input_shape)
+        pre_pad_shape[curr_dim] = pre_pad
+        pre_pad_tensor = torch.full(pre_pad_shape, float(value))
```

Comment:
Due to the `FakeTensor` issue in `torch.compile`, could this instead be a Numpy array?